src/notation/view/percussionpanel/percussionpanelpadlistmodel.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-Studio-CLA-applies
+ *
+ * MuseScore Studio
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2024 MuseScore Limited
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "percussionpanelpadlistmodel.h"
+
+PercussionPanelPadListModel::PercussionPanelPadListModel(QObject* parent)
+    : QAbstractListModel{parent}
+{
+    m_padModels = defaultItems();
+}
+
+void PercussionPanelPadListModel::addRow()
+{
+    for (size_t i = 0; i < NUM_COLUMNS; ++i) {
+        m_padModels.append(new PercussionPanelPadModel(QObject::parent()));
+    }
+    emit layoutChanged();
```

Comment:
We need to use something that allows us to update the view more locally (for example, begin/endRemoveRows)